jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/URIUtil.java

```diff
@@ -758,7 +759,7 @@ public static String canonicalPath(String encodedPath)
                     default:
                         if (builder == null && !isSafe(c))
                         {
-                            builder = new Utf8StringBuilder(encodedPath.length());
+                            builder = new Utf8StringBuilder(encodedPath.length(), CodingErrorAction.REPLACE);
```

Comment:
Interesting that we were not falling back to 8859-1 in this method, so we are somewhat inconsistent.    As we pretty much always call canonicalPath, then we pretty much always threw.   Now we will replace. Hmm back to being cautious about a big behaviour change.